tests/api_tests/nodes/test_views.py

```diff
@@ -689,6 +697,44 @@ def test_node_update_invalid_data(self):
         assert_equal(res.status_code, 400)
         assert_equal(res.json['errors'][0]['detail'], "Malformed request.")
 
+    @assert_not_logs(NodeLog.MADE_PUBLIC, 'private_project')
+    def test_cannot_make_project_public_if_non_contributor(self):
+        non_contrib = AuthUserFactory()
+        res = self.app.patch_json(
+            self.private_url,
+            make_node_payload(self.private_project, {'public': True}),
+            auth=non_contrib.auth, expect_errors=True
+        )
+        assert_equal(res.status_code, 403)
+
+    def test_cannot_make_project_public_if_non_admin_contributor(self):
+        non_admin = AuthUserFactory()
+        self.private_project.add_contributor(
+            non_admin,
+            permissions=(permissions.READ, permissions.WRITE),
+            auth=Auth(self.private_project.creator)
+        )
+        self.private_project.save()
+        res = self.app.patch_json(
+            self.private_url,
+            make_node_payload(self.private_project, {'public': True}),
+            auth=non_admin.auth, expect_errors=True
+        )
+        assert_equal(res.status_code, 403)
+
+        self.private_project.reload()
+        assert_false(self.private_project.is_public)
+
+    @assert_logs(NodeLog.MADE_PUBLIC, 'private_project')
+    def test_can_make_project_public_if_admin_contributor(self):
+        res = self.app.patch_json_api(
+            self.private_url,
+            make_node_payload(self.private_project, {'public': True}),
+            auth=self.user.auth  # self.user is creator/admin
```

Comment:
Fair enough. Was trying to avoid creating extra `AuthUserFactories`, as they somewhat slow down the tests (due to having to bcrypt-hash passwords). But it's not a big deal--readability is more important. Will make the change.